### ISSUE TITLE
createrepo_c 0.13.0 unit tests fail when libmodulemd support is not enabled

### ISSUE THREAD

**Issue by Conan-Kudo** [rpm-software-management/createrepo_c#153]
When createrepo_c support for libmodulemd is not enabled, the unit tests fail because the test data now includes modules.yaml content.

```
[   10s] test_xml_parser_repomd_repo01_nowarningcb (tests.test_xml_parser.TestCaseXmlParserRepomd) ... ok
[   10s]
[   10s] ======================================================================
[   10s] ERROR: test_metadatalocation (tests.test_locate_metadata.TestCaseMetadataLocation)
[   10s] ----------------------------------------------------------------------
[   10s] Traceback (most recent call last):
[   10s]   File "/home/abuild/rpmbuild/BUILD/createrepo_c-0.13.0/tests/python/tests/test_locate_metadata.py", line 28, in test_metadatalocation
[   10s]     ml = cr.MetadataLocation(REPO_WITH_ADDITIONAL_METADATA, 0)
[   10s] createrepo_c.CreaterepoCError: Module metadata found in repository, but createrepo_c was not compiled with libmodulemd support.
[   10s]
[   10s] ----------------------------------------------------------------------
[   10s] Ran 97 tests in 0.283s
[   10s]
[   10s] FAILED (errors=1)
```

This test failure was introduced by fab3e4a183adad30577a42f2f7419802d5a42022.

**Comment by Conan-Kudo** [comment]
This probably implies that bombing out on module metadata being detected might be a bad idea, too. I'm not completely sure about this, though.

**Comment by Conan-Kudo** [comment]
cc: @sgallagher, @dmach, @kontura.

**Comment by sgallagher** [comment]
That test appears to have been added *after* the modulemd-awareness patches, so I think @kontura likely didn't run the tests in both build configurations.

FWIW, that behavior is expected. If we see a repo with modulemd and we don't have libmodulemd support compiled in, there's no safe fallback. Exiting with an error is the safest option.

**Comment by kontura** [comment]
I have actually ran the tests even with libmodulemd disabled, but I left it like this on purpose. I though that it's expected the tests should fail. Only now do I realize that we probably need those tests to pass in order to build it with libmodulemd disabled? So I should have used something like `#ifdef WITH_LIBMODULEMD`, my bad.

**Comment by sgallagher** [comment]
Better still would be for it to run the test and verify that it gets that error back.

**Comment by kontura** [comment]
Yes, that's what I meant, but since the test should pass when we have libmodulemd and fail if we don't, there needs to be some way of knowing which case is the current run. I am just now not sure how to get this information into the Python test case.

**Comment by sgallagher** [comment]
> Yes, that's what I meant, but since the test should pass when we have libmodulemd and fail if we don't, there needs to be some way of knowing which case is the current run. I am just now not sure how to get this information into the Python test case.

Well, CMake knows, so can't you just set an environment variable for the python to check?

**Comment by Conan-Kudo** [comment]
Fixed with the release of v0.13.1.
